### PR TITLE
4853-renaming resource type names for the UI

### PR DESCRIPTION
**Pull request by pkdash**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Verify the resource Create options are: RESOURCE, COLLECTION, and APP CONNECTOR
2. Create a RESOURCE. Verify on landing page of the resource the 'Type' is 'Resource'. Add an aggregation to this resource. Make the resource discoverable.
3. Create an APP CONNECTOR. Verify the landing page of the connector the 'Type' is "App Connector". In edit mode of the landing page, verify the supported types (checkbox) are: Resource, and Collection. Make the connector discoverable.
4. Create a COLLECTION. Verify on landing page of the collection the 'Type' is 'Collection'. Add the resource to this collection. Verify the Add resource dialog shows the 'Type' as 'Resource', 'Collection', and 'App Connector'. Make the collection discoverable. Verify the type for the added resources in collection resource are 'Resource', 'App Connector'.
5. Go to HOME page. Verify that the 3 resources have the "Type" listed as "Resource", "Collection", and "App Connector".
6. Go to MY RESOURCES page. Verify the resource icons tooltip shows type as "Resource", "Collection", and "App Connector" for the respective icon. In the search bar dropdown, under the Type, verify that the options are: All, Resource, Collection, App Connector. 
7. Go to DISCOVER page. Verify the tooltip for the 3 resources are as in step 6 above. Verify the 'Content Type' filter section.
8. Create a group. Give resource access permission to the group for the resource, collection and app connector. Go to the group page. Verify under the 'Resources' tab the resource icon tooltip are one of these: Resource, Collection or App Connector. 
9. Go to profile page. Under the 'OVERVIEW' tab, verify the 'Type'. Under the 'CONTRIBUTIONS'  tab, verify that only 'Resource', 'Collection' and 'App Connector' are listed.

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4874](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4874)

File: hs_collection_resource/templates/pages/collectionresource.html [hunk 1]
```diff
@@ -211,7 +219,15 @@ <h4 class="modal-title">Deleted Resources</h4>
                                       {% endwith %}
                                     </td>
                                     <td>
-                                        {{ res.resource_type }}
+                                        {% if res.resource_type == "CompositeResource" %}
+                                            Resource
+                                        {% elif res.resource_type == "CollectionResource" %}
+                                            Collection
+                                        {% elif res.resource_type == "ToolResource" %}
+                                            Web App Connector
```
Comment (devincowan) [inline hs_collection_resource/templates/pages/collectionresource.html:227]: "App Connector" ?

File: hs_collection_resource/templates/pages/collectionresource.html [hunk 2]
```diff
@@ -55,7 +55,15 @@ <h3>Collection Contents </h3>
                                             {% endif %}
                                         </td>
                                         <td>
-                                            {{ res.resource_type }}
+                                            {% if res.resource_type == "CompositeResource" %}
+                                                Resource
```
Comment (devincowan) [inline hs_collection_resource/templates/pages/collectionresource.html:59]: unsure if we want to change "composite resource" to "resource" here:
https://github.com/hydroshare/hydroshare/blob/4853-resource-type-name-fix/hs_composite_resource/views.py#L45

File: hs_core/models.py [hunk 3]
```diff
@@ -3769,8 +3770,13 @@ def verbose_name(self):
 
     @property
     def discovery_content_type(self):
-        """Return verbose name of content type."""
-        return self.get_content_model().discovery_content_type
+        """Return name used for the content type in discovery/solr search."""
+        return self.get_content_model().display_name
+
+    # @property
```
Comment (devincowan) [inline hs_core/models.py:3776]: intentional comment?

File: hs_discover/src/components/Search.vue [hunk 4]
```diff
@@ -293,9 +293,9 @@ export default {
       countAvailabilities: {},
       availabilityFilter: [],
       resIconName: {
-        'Composite Resource': '/static/static/img/resource-icons/composite48x48.png',
-        'Collection Resource': '/static/static/img/resource-icons/collection48x48.png',
-        'Web App Resource': '/static/static/img/resource-icons/webapp48x48.png',
+        Resource: '/static/static/img/resource-icons/composite48x48.png',
```
Comment (devincowan) [inline hs_discover/src/components/Search.vue:296]: missing ''

File: theme/templates/includes/navbar.html [hunk 5]
```diff
@@ -337,12 +337,12 @@ <h4>Notifications <span v-if="tasks.length">(${tasks.length})</span></h4>
                                 <li>
                                     <a data-toggle="tooltip" data-placement="left"
                                        title="A shortcut (or reference) to an external web application that functions with HydroShare data resources."
-                                       data-modal-title="Create a Web App Connector"
-                                       data-modal-input-title="Enter the name for the new Web App Connector."
+                                       data-modal-title="Create a App Connector"
+                                       data-modal-input-title="Enter the name for the new app connector."
```
Comment (devincowan) [inline theme/templates/includes/navbar.html:341]: caps?

File: theme/templatetags/dashboard_tags.py [hunk 6]
```diff
@@ -51,13 +51,11 @@ def resource_verbose_name_builder(rsc_type):
...
         'CompositeResource': 'composite',
-        'Time Series': 'timeseries',
         'ToolResource': 'webapp',
         'CollectionResource': 'collection'
     }
     dict_verbose_name = {
-        'CompositeResource': 'Composite',
-        'Time Series': 'Time Series',
-        'ToolResource': 'Web App',
+        'CompositeResource': 'Resource',
```
Comment (devincowan) [inline theme/templatetags/dashboard_tags.py:58]: do we need to edit Types here:
https://github.com/hydroshare/hydroshare/blob/4853-resource-type-name-fix/theme/templates/pages/group-unauthenticated.html#L195

File: theme/templates/resource-landing-page/title-section.html [hunk 7]
```diff
@@ -5,14 +5,14 @@
     <div class="col-xs-12">
         <div class="alert alert-warning alert-dismissible" role="alert">
             <button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button>
-            <span><strong>This resource is under review</strong> for appropriate 
-                <a href="https://help.hydroshare.org/publishing-in-hydroshare/minimum-metadata-requirements-for-publishing-in-hydroshare/" target="_blank">minimum metadata</a> 
-                and to ensure that it adheres to 
+            <span><strong>This resource is under review</strong> for appropriate
```
Comment (devincowan) [inline theme/templates/resource-landing-page/title-section.html:8]: thank you for this

File: hs_core/templates/pages/my-resources.html [hunk 8]
```diff
@@ -96,23 +96,12 @@ <h4 class="panel-title">
                                     <div class="dropdown-menu dropdown-menu-left shadow-md" role="menu">
                                         <form class="form-horizontal" role="form">
                                             <div class="form-group">
-                                                <label for="filter">Resource Type</label>
+                                                <label for="filter">Type</label>
                                                 <select id="input-resource-type" class="form-control pull-left">
                                                     <option value="" selected>All</option>
                                                     <option value="collection resource">Collection</option>
-                                                    <option value="composite resource">Composite Resource</option>
-                                                    <option value="generic">Generic</option>
```
Comment (devincowan) [inline hs_core/templates/pages/my-resources.html:104]: "Generic resource" here need to change?
https://github.com/hydroshare/hydroshare/blob/4853-resource-type-name-fix/hs_core/templates/pages/share-resource.html#L12

File: hs_core/templates/pages/my-resources.html [hunk 9]
```diff
@@ -96,23 +96,12 @@ <h4 class="panel-title">
...
-                                                    <option value="geographic feature">Geographic Feature (ESRI Shapefiles)</option>
-                                                    <option value="geographic raster">Geographic Raster</option>
-                                                    <option value="his referenced time series">HIS Referenced Time Series</option>
-                                                    <option value="model instance resource">Model Instance</option>
-                                                    <option value="model program">Model Program</option>
-                                                    <option value="modflow model instance resource">MODFLOW Model Instance</option>
-                                                    <option value="multidimensional">Multidimensional (NetCDF)</option>
-                                                    <option value="script resource">Script Resource</option>
```
Comment (devincowan) [inline hs_core/templates/pages/my-resources.html:112]: do we need to update the resource types here:
https://github.com/hydroshare/hydroshare/blob/4853-resource-type-name-fix/theme/templates/pages/group-unauthenticated.html#L189

**Review by devincowan — CHANGES_REQUESTED** [review]
I love this simplification! A few small items and as always, some questions

File: hs_collection_resource/templates/pages/collectionresource.html (hunk 2 above)
Comment (pkdash) [reply hs_collection_resource/templates/pages/collectionresource.html:59]: I was thinking of changing it and thought it would require probably changing at many places where we are raising error related to composite resource. I guess since these error messages are for the user, I should change it.

File: hs_core/models.py (hunk 3 above)
Comment (pkdash) [reply hs_core/models.py:3776]: Forgot to delete the commented code.

File: hs_discover/src/components/Search.vue (hunk 4 above)
Comment (pkdash) [reply hs_discover/src/components/Search.vue:296]: Since the key name doesn't have space quotes not needed.

File: theme/templates/includes/navbar.html (hunk 5 above)
Comment (pkdash) [reply theme/templates/includes/navbar.html:341]: I changed it to lower case to be consistent with resource and collection modals.

File: hs_core/templates/pages/my-resources.html (hunk 8 above)
Comment (pkdash) [reply hs_core/templates/pages/my-resources.html:104]: Yeah need to clean all the old resource types here.

File: hs_core/templates/pages/my-resources.html (hunk 9 above)
Comment (pkdash) [reply hs_core/templates/pages/my-resources.html:112]: Yeah. Thanks for catching this.

File: hs_discover/src/components/Search.vue (hunk 4 above)
Comment (devincowan) [reply hs_discover/src/components/Search.vue:296]: right! nice!

File: hs_collection_resource/templates/pages/collectionresource.html (hunk 2 above)
Comment (devincowan) [reply hs_collection_resource/templates/pages/collectionresource.html:59]: all good either way for me, just checking...

File: theme/templates/includes/navbar.html (hunk 5 above)
Comment (devincowan) [reply theme/templates/includes/navbar.html:341]: makes sense

File: hs_core/templates/pages/my-resources.html (hunk 8 above)
Comment (pkdash) [reply hs_core/templates/pages/my-resources.html:104]: Good question. Not sure where/how this template is being used.

File: theme/templatetags/dashboard_tags.py (hunk 6 above)
Comment (pkdash) [reply theme/templatetags/dashboard_tags.py:58]: We don't need to change it here.

File: hs_core/templates/pages/my-resources.html (hunk 8 above)
Comment (devincowan) [reply hs_core/templates/pages/my-resources.html:104]: Good call, I don't see that it is used...

File: hs_collection_resource/templates/pages/collectionresource.html (hunk 2 above)
Comment (pkdash) [reply hs_collection_resource/templates/pages/collectionresource.html:59]: I have updated those messages where appropriate.

Comment (devincowan) [reply hs_collection_resource/templates/pages/collectionresource.html:59]: Nice!

**Comment by pkdash** [comment]
@devincowan See I have addressed your comments.

File: hs_core/views/resource_folder_hierarchy.py [hunk 10]
```diff
@@ -318,7 +318,8 @@ def zip_aggregation_file(request, res_id=None):
         return JsonResponse({"error": "Permission denied"}, status=status.HTTP_401_UNAUTHORIZED)
 
     if resource.resource_type != "CompositeResource":
-        return JsonResponse({"error": "Resource is not of type composite resource"}, status=status.HTTP_400_BAD_REQUEST)
+        err_msg = f"{resource.display_name} type doesn't support zipping of aggregation."
```
Comment (devincowan) [inline hs_core/views/resource_folder_hierarchy.py:321]: Unzipping within the file browser failed for me locally, within a standard Composite resource. Not sure if this was something about my build, so worth testing on your end.
image

File: theme/static/js/resourceLandingAjax.js [hunk 11]
```diff
@@ -195,14 +195,14 @@ function metadata_update_ajax_submit(form_id){
...
                     ((json_response.element_name.toLowerCase() === 'coverage' ||
-                    json_response.element_name.toLowerCase() === 'site') && resourceType === 'Composite Resource')){
+                    json_response.element_name.toLowerCase() === 'site') && resourceType === 'Resource')){
                     if (json_response.hasOwnProperty('temporal_coverage')){
                         var temporalCoverage = json_response.temporal_coverage;
                         updateResourceTemporalCoverage(temporalCoverage);
                         // show/hide delete option for resource temporal coverage
                         setResourceTemporalCoverageDeleteOption();
 
-                        if(resourceType === 'Composite Resource' && json_response.has_logical_temporal_coverage) {
```
Comment (devincowan) [inline theme/static/js/resourceLandingAjax.js:205]: does this "Composite Resource" need a change?
https://github.com/hydroshare/hydroshare/blob/4853-resource-type-name-fix/hs_core/management/commands/set_aggregations_dirty.py#L55

**Review by devincowan — APPROVED** [review]
looks nice and clean! a couple things to check quick but otherwise good to go

File: theme/static/js/resourceLandingAjax.js (hunk 11 above)
Comment (pkdash) [reply theme/static/js/resourceLandingAjax.js:205]: I think change is not needed since this is in a management command user doesn't see it.
